src/components/reservation/ReservationModalBody.tsx

```diff
@@ -74,32 +65,27 @@ const ReservationModalBody = ({
   const branches = config<AgencyBranch[]>("branchesConfig", {
     transformer: "jsonParse"
   });
-
-  const mainManifestationType = getManifestationType(mainManifestation);
+  const mainManifestationType = getManifestationType(selectedManifestations[0]);
   const { reservableManifestations } = UseReservableManifestations({
-    manifestations:
-      // TODO: We should investigate why we need to check for parallelManifestation length
-      // because it doesn't seem to reflect the possible types.
-      parallelManifestations && parallelManifestations.length > 0
-        ? parallelManifestations
-        : [mainManifestation],
+    manifestations: selectedManifestations,
     type: mainManifestationType
   });
   const queryClient = useQueryClient();
   const [reservationResponse, setReservationResponse] =
     useState<ReservationResponseV2 | null>(null);
   const [selectedBranch, setSelectedBranch] = useState<string | null>(null);
   const [selectedInterest, setSelectedInterest] = useState<number | null>(null);
-  const faustId = convertPostIdToFaustId(pid);
+  const allPids = getAllPids(selectedManifestations);
+  const faustIds = allPids.map((pid) => convertPostIdToFaustId(pid));
   const { mutate } = useAddReservationsV2();
   const userResponse = useGetPatronInformationByPatronIdV2();
   const holdingsResponse = useGetHoldingsV3({
-    recordid: [faustId]
+    recordid: faustIds
```

Comment:
Please follow through with this change. 

We pass an array of ids but on L100 we only work with the first holdings result. That does not seem right.